hive/src/main/java/org/apache/iceberg/hive/HiveTableOperations.java

```diff
@@ -231,8 +242,11 @@ private long acquireLock() throws UnknownHostException, TException {
   private void unlock(Optional<Long> lockId) {
     if (lockId.isPresent()) {
       try {
-        metaStoreClient.unlock(new UnlockRequest(lockId.get()));
-      } catch (TException e) {
+        metaClients.run(client -> {
+          client.unlock(lockId.get());
+          return null;
+        });
+      } catch (Exception e) {
```

Comment:
TODO: This should be `InterruptedException`

Reply:
I was wrong. This needs to be Exception to include the TException thrown if unlock fails.